contracts/collectors/Pool.sol

```diff
@@ -148,20 +150,22 @@ contract Pool is IPool, OwnablePausableUpgradeable {
             return;
         }
 
-        IStakedEthToken _stakedEthToken = stakedEthToken; // gas savings
-        uint256 totalSupply = _stakedEthToken.totalSupply();
-        if (totalSupply == 0) {
-            _stakedEthToken.mint(msg.sender, msg.value);
+        uint256 _totalStakingAmount = totalStakingAmount; // gas savings
+        if (_totalStakingAmount == 0) {
+            stakedEthToken.mint(msg.sender, msg.value);
             return;
         }
 
+        // calculate total activating amount
+        uint256 totalActivatingAmount = newTotalCollectedAmount.sub(_totalStakingAmount);
+
         // calculate activating share
         // multiply by 10000 as minActivatingShare is stored  up to 10000 (5.25% -> 525)
-        uint256 activatingShare = newTotalActivatingAmount.mul(1e22).div(totalSupply);
+        uint256 activatingShare = totalActivatingAmount.mul(1e22).div(newTotalCollectedAmount);
```

Comment:
I think the divisor should be `_totalStakingAmount` instead of `newTotalCollectedAmount`.